Kickstarter-iOS/Features/PledgeView/Controllers/PostCampaignCheckoutViewController.swift

```diff
@@ -402,38 +403,83 @@ extension PostCampaignCheckoutViewController: PledgeViewControllerMessageDisplay
   }
 }
 
-// MARK: - MessageBannerViewControllerDelegate
+// MARK: - STPApplePayContextDelegate
 
-extension PostCampaignCheckoutViewController: PKPaymentAuthorizationViewControllerDelegate {
-  func paymentAuthorizationViewControllerDidFinish(_ controller: PKPaymentAuthorizationViewController) {
-    controller.dismiss(animated: true, completion: { [weak self] in
-      self?.viewModel.inputs.paymentAuthorizationViewControllerDidFinish()
-    })
-  }
+enum PostCampaignCheckoutApplePayError: Error {
+  case missingToken(String)
+  case missingPaymentMethodInfo(String)
+  case missingPaymentIntent(String)
+}
 
-  func paymentAuthorizationViewController(
-    _: PKPaymentAuthorizationViewController,
-    didAuthorizePayment payment: PKPayment,
-    handler completion: @escaping (PKPaymentAuthorizationResult)
-      -> Void
+extension PostCampaignCheckoutViewController: STPApplePayContextDelegate {
+  func applePayContext(
+    _: StripeApplePay.STPApplePayContext,
+    didCreatePaymentMethod _: StripePayments.STPPaymentMethod,
+    paymentInformation payment: PKPayment,
+    completion: @escaping StripeApplePay.STPIntentClientSecretCompletionBlock
   ) {
-    let paymentDisplayName = payment.token.paymentMethod.displayName
-    let paymentNetworkName = payment.token.paymentMethod.network?.rawValue
+    guard let paymentIntentClientSecret = self.applePayPaymentIntent else {
+      completion(
+        nil,
+        PostCampaignCheckoutApplePayError.missingPaymentIntent("Missing PaymentIntent")
+      )
+      return
+    }
+
+    guard let paymentDisplayName = payment.token.paymentMethod.displayName,
+      let paymentNetworkName = payment.token.paymentMethod.network?.rawValue else {
+      completion(
+        nil,
+        PostCampaignCheckoutApplePayError
+          .missingPaymentMethodInfo("Unable to retrieve payment method information from ApplePay")
+      )
+      return
+    }
+
     let transactionId = payment.token.transactionIdentifier
 
-    self.viewModel.inputs.paymentAuthorizationDidAuthorizePayment(paymentData: (
-      paymentDisplayName,
-      paymentNetworkName,
-      transactionId
-    ))
+    // Tokens considered 'legacy' from Stripe's side, but we still store the token ourselves.
+    STPAPIClient.shared.createToken(with: payment) { [weak self] token, _ in
```

Comment:
This is the one manual call we make to Stripe, and it's because our backend requires the token.